src/poaps/domain/POAP.ts

```diff
@@ -51,17 +69,67 @@ export class POAP {
   }
 }
 
-interface PoapProperties {
-  id: number;
-  collectorAddress: string;
-  transferCount: number;
-  mintedOn: Date;
-  dropId: number;
-  imageUrl: string;
-  city: string;
-  country: string;
-  description: string;
-  startDate: Date;
-  endDate: Date;
-  name: string;
+export interface POAPMintingStats {
+  mintOrder: number;
+}
+
+export interface POAPCollectorStats {
+  power: number;
+}
+
+export interface POAPDropStats {
+  poapCount: number;
+}
+
+interface PoapWithStatsProperties extends PoapProperties {
+  mintingStats?: POAPMintingStats;
+  collectorStats?: POAPCollectorStats;
+  dropStats?: POAPDropStats;
+}
+
+export class POAPWithStats extends POAP {
+  readonly mintingStats?: POAPMintingStats;
+  readonly collectorStats?: POAPCollectorStats;
+  readonly dropStats?: POAPDropStats;
+
+  constructor(properties: PoapWithStatsProperties) {
+    super(properties);
+
+    this.mintingStats = properties.mintingStats;
+    this.collectorStats = properties.collectorStats;
+    this.dropStats = properties.dropStats;
+  }
+
+  // eslint-disable-next-line complexity
+  public static fromCompass(response: PoapsWithStatsResponse): POAPWithStats {
+    const mintedOn = new Date(0);
+    mintedOn.setUTCSeconds(response.minted_on);
+
+    return new POAPWithStats({
+      id: Number(response.id),
+      collectorAddress: response.collector_address,
+      transferCount: response.transfer_count,
+      mintedOn,
+      dropId: Number(response.drop_id),
+      imageUrl: response.drop.image_url,
+      city: response.drop.city,
+      country: response.drop.country,
+      description: response.drop.description,
+      startDate: new Date(response.drop.start_date),
+      endDate: new Date(response.drop.end_date),
+      name: response.drop.name,
+      mintingStats: response.minting_stats?.mint_order
+        ? { mintOrder: response.minting_stats.mint_order }
+        : undefined,
+      collectorStats: response.collector?.poaps_owned
```

Comment:
0 is a valid number of POAPs owned.
```suggestion
      collectorStats: response.collector?.poaps_owned != null
```